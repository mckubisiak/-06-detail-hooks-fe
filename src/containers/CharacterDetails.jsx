import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import Character from '../components/Character';
import { fetchCharacterById } from '../services/fetchApi';

const CharacterDetails = () => {
  const { id } = useParams();
  const [loading, setLoading] = useState(true);
  const [chartacter, setCharacter] = useState({});

  useEffect(() => {
    const loadDetails = async () => {
      const charcter = await fetchCharacterById(id);
      setCharacter(charcter);
      setLoading(false);
    };
    loadDetails();
  }, [id]);

  if (loading) return (
    <img src="https://i.redd.it/o6m7b0l6h6pz.gif" alt="loading spinner" />);
  
  return <Character {...chartacter} />;
};

export default CharacterDetails;
