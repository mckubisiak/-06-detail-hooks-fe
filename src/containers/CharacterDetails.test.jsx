import React from 'react';
import { screen, render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import mortydetail from '../fixtures/mortydetail.json';
import CharacterDetails from './CharacterDetails';

const mockServer = setupServer(
  rest.get('https://rickandmortyapi.com/api/character/:id', (req, res, ctx) => {
    return res(ctx.json(mortydetail));
  })
);

describe('chartater detail pag', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());

  it('has the details of only a single character', async () => {
    render(
      <MemoryRouter initialEntries={['/2']}>
        <CharacterDetails />
      </MemoryRouter>
    );

    screen.getByAltText('loading spinner');

    const character = await screen.findByText('Morty Smith', { exact: false });

    expect(character).toMatchSnapshot();
  });
});
