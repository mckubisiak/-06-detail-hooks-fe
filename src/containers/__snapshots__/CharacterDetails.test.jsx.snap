// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chartater detail pag has the details of only a single character 1`] = `
<a
  href="/2"
>
  Morty Smith
</a>
`;
